semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala

```diff
@@ -75,7 +75,10 @@ trait SymbolOps { self: SemanticdbOps =>
     def isSemanticdbMulti: Boolean = sym.isOverloaded
     def disambiguator: String = {
       val peers = sym.owner.semanticdbDecls.gsyms
-      val overloads = peers.filter(_.name == sym.name)
+      val overloads = peers.filter { peer =>
+        peer.isMethod &&
+        peer.name == sym.name
+      }
```

Comment:
What about `object foo; def foo(x: Int) = ???`? Will code like this be handled correctly?

Reply:
Sounds good.